### PR TITLE
Update karl's languages to use test runner v3 features

### PR DESCRIPTION
**Pull request by aldotele**
This PR addresses karl's languages task on issue #2356 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 1]
```diff
@@ -1,44 +1,32 @@
...
-import org.junit.Test;
+import static org.assertj.core.api.Assertions.*;
 
 public class LanguageListTest {
 
     LanguageList languageList = new LanguageList();
 
     @Test
+    @Tag("task:1")
+    @DisplayName("The isEmpty method returns true when no language was added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:13]:
```suggestion
    @DisplayName("The isEmpty method returns true when the list contains no languages")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 2]
```diff
@@ -1,44 +1,32 @@
...
     public void empty() {
         assertThat(languageList.isEmpty()).isTrue();
     }
 
     @Test
+    @Tag("task:2")
+    @DisplayName("The addLanguage method correctly adds a language to the list so that list is no more empty")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:20]:
```suggestion
    @DisplayName("The isEmpty method returns false after adding a language to the list")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 3]
```diff
@@ -1,44 +1,32 @@
...
-
-        assertThat(languageList.containsLanguage("Java")).isTrue();
-        assertThat(languageList.containsLanguage("Ruby")).isTrue();
-        assertThat(languageList.containsLanguage("C++")).isTrue();
-        assertThat(languageList.containsLanguage("Python")).isFalse();
-    }
-
-    @Test
+    @Tag("task:3")
+    @DisplayName("The removeLanguage method correctly removes a language from the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:29]: This test is a bit weird: it uses `containsLanguage` to check whether the language was removed, but this method is not yet implemented in task 3.

I would suggest refactoring the test method to only use methods implemented so far, by making it

1. Add a language
2. Remove that same language
3. Asserting that the list is empty again.

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 4]
```diff
@@ -52,6 +40,8 @@ public void removeLanguage() {
     }
 
     @Test
+    @Tag("task:4")
+    @DisplayName("The firstLanguage method correctly retrieves the first language that was added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:44]:
```suggestion
    @DisplayName("The firstLanguage method returns the first language that was added to the list")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 5]
```diff
@@ -61,6 +51,8 @@ public void firstLanguage() {
     }
 
     @Test
+    @Tag("task:5")
+    @DisplayName("The count method returns the correct number of languages that were added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:55]:
```suggestion
    @DisplayName("The count method returns the number of languages in the list")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 6]
```diff
@@ -70,18 +62,48 @@ public void countThree() {
     }
 
     @Test
+    @Tag("task:5")
+    @DisplayName("The count method returns 0 in case no language was added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:66]:
```suggestion
    @DisplayName("The count method returns 0 when the list is empty")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 7]
```diff
@@ -70,18 +62,48 @@ public void countThree() {
...
+    public void addMultipleLanguages() {
+        languageList.addLanguage("Java");
+        languageList.addLanguage("Ruby");
+        languageList.addLanguage("C++");
+
+        assertThat(languageList.containsLanguage("Java")).isTrue();
+        assertThat(languageList.containsLanguage("Ruby")).isTrue();
+        assertThat(languageList.containsLanguage("C++")).isTrue();
+        assertThat(languageList.containsLanguage("Python")).isFalse();
+    }
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:93]: I would suggest refactoring these tests to make them a bit simpler, something like this:

```java
@Test
@Tag("task:6")
@DisplayName("The containsLanguage returns true when the language is in the list")
public void containsLanguage() {
     languageList.addLanguage("Java");

    assertThat(languageList.containsLanguage("Java")).isTrue();
}

@Test
@Tag("task:6")
@DisplayName("The containsLanguage returns false when the language is not in the list")
public void doesNotContainLanguage() {
    languageList.addLanguage("Kotlin");

    assertThat(languageList.containsLanguage("Java")).isFalse();
}
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 8]
```diff
@@ -70,18 +62,48 @@ public void countThree() {
...
+
+    @Test
+    @Tag("task:7")
+    @DisplayName("The isExciting method returns true after Java was added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:97]:
```suggestion
    @DisplayName("The isExciting method returns true when the list contains Java")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 9]
```diff
@@ -70,18 +62,48 @@ public void countThree() {
...
     public void excitingLanguageListWithJava() {
         languageList.addLanguage("Java");
 
         assertThat(languageList.isExciting()).isTrue();
     }
 
     @Test
+    @Tag("task:7")
+    @DisplayName("The isExciting method returns true after Kotlin was added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:106]:
```suggestion
    @DisplayName("The isExciting method returns true when the list contains Kotlin")
```

File: exercises/concept/karls-languages/src/test/java/LanguageListTest.java [hunk 10]
```diff
@@ -90,6 +112,8 @@ public void excitingLanguageListWithKotlin() {
     }
 
     @Test
+    @Tag("task:7")
+    @DisplayName("The isExciting method returns false when neither Java nor Kotlin were added to the list")
```
Comment (sanderploegsma) [inline exercises/concept/karls-languages/src/test/java/LanguageListTest.java:116]:
```suggestion
    @DisplayName("The isExciting method returns false when the list contains neither Java nor Kotlin")
```

**Review by sanderploegsma — CHANGES_REQUESTED** [review]
I have left a bunch of comments this time. Most of them are suggestions on the naming of the tests, as I believe they can be a bit shorter and clearer.

The others are about the test cases themselves, I believe we can make them a bit better.

**Comment by aldotele** [comment]
Just commited the enhancements, thank you for the review!

**Review by sanderploegsma — APPROVED** [review]
LGTM, thanks!

**Pull request merged**